sky/skylet/job_lib.py

```diff
@@ -503,21 +502,22 @@ def cancel_jobs(job_owner: str, jobs: Optional[List[int]]) -> None:
     else:
         job_records = _get_jobs_by_ids(jobs)
 
-    jobs = [make_ray_job_id(job['job_id'], job_owner) for job in job_records]
     # TODO(zhwu): `job_client.stop_job` will wait for the jobs to be killed, but
     # when the memory is not enough, this will keep waiting.
     job_client = _create_ray_job_submission_client()
 
-    def stop_job(job: str):
+    # Sequentially cancel the jobs to avoid the resource number bug caused by
+    # ray cluster (tracked in #1262).
+    for job in job_records:
```

Comment:
Do `job_records` contain jobs that are already finished or cancelled?
If so, should we filter them out to avoid potential overheads?

Reply:
Yes, it will only obtain the jobs that are PENDING or RUNNING, if the user specifies `sky cancel -a`. If the user specifies the id explicitly, we will still run the stop for that job, which I think should be fine, since that is a redundant way to cancel the job.